components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice.go

```diff
@@ -0,0 +1,220 @@
+	CheckPath      func(cmd string) (string, error)
+}
+
+func NewSoftwareVersionService(logger logger.Logger, CheckPath func(cmd string) (string, error)) ISoftwareVersionService {
+	return &SoftwareVersionService{
+		cmdCheckArray:  cmdCheckArray,
+		osFilePath:     OSFILEPATH,
+		kernelFilePath: KERNELFILEPATH,
+		logger:         logger,
+	}
+}
+
+const (
+	AVAILABILITY              = " availability"
+	ENSURE                    = "Ensure "
+	OSFILEPATH                = "/etc/os-release"
+	KERNELFILEPATH            = "/proc/sys/kernel/osrelease"
+	LINUX_VERSION_CHECK       = "Linux Version Check"
+	KERNAL_VERSION_CHECK      = "Kernal Version Check"
+	UBUNTU                    = "Ubuntu"
+	RED_HAT                   = "Red Hat"
+	OPENSEARCH                = "opensearch"
+	AUTOMATE                  = "automate"
+	POSTGRES                  = "postgres"
+	CHEF_SERVER               = "chef-server"
+	BASTION                   = "bastion"
+	DEBIAN                    = "Debian"
+	RED_HAT_SUPPORTED_VERSION = 7
+	KERNAL_SUPPORTED_VERSION  = 3.2
+)
+
+var cmdCheckArray = []string{"mkdir", "useradd", "chown", "rm", "touch", "truncate", "echo", "sleep", "ls", "grep", "yum", "which", "cp", "curl", "bash", "sysctl", "cat", "sed", "mount", "mv", "systemd", "wget", "exec", "rsync"}
+
+func (sv *SoftwareVersionService) GetSoftwareVersionDetails(query string) (*models.SoftwareVersionDetails, error) {
+	sv.logger.Debug("The query parameter entered: ", query)
+	serviceResponse := &models.SoftwareVersionDetails{}
+	serviceResponse.Passed = true
+	serviceResponseArray := []*models.Checks{}
+	cmdArray := []string{}
+	switch query {
+	case POSTGRES:
+		cmdArray = []string{"stat"}
+		cmdArray = append(cmdArray, sv.cmdCheckArray...)
+	case OPENSEARCH:
+		cmdArray = []string{"openssl"}
+		cmdArray = append(cmdArray, sv.cmdCheckArray...)
+	case AUTOMATE, CHEF_SERVER, BASTION:
+		cmdArray = append(cmdArray, sv.cmdCheckArray...)
+	default:
+		sv.logger.Error("The Query parameter is not supported")
+		return nil, errors.New("The query " + query + " is not supported. The Supported query's are: postgres, opensearch, bastion, automate, chef-server")
+	}
+
+	for i := 0; i < len(cmdArray); i++ {
+		checkResponse := sv.checkCommandVersion(cmdArray[i])
+		if !checkResponse.Passed {
+			serviceResponse.Passed = false
+		}
+		serviceResponseArray = append(serviceResponseArray, checkResponse)
+	}
+	osResponse, err := sv.checkOsVersion(sv.osFilePath)
+	if err != nil {
+		sv.logger.Error("Error while getting the OS Version: ", osResponse.ErrorMsg)
+	}
+	if !osResponse.Passed {
+		serviceResponse.Passed = false
+	}
+	kernelResponse, err := sv.checkKernelVersion(sv.kernelFilePath)
+	if err != nil {
+		sv.logger.Error("Error while getting the Kernal Version: ", kernelResponse.ErrorMsg)
+	}
+	if !kernelResponse.Passed {
+		serviceResponse.Passed = false
+	}
+	serviceResponseArray = append(serviceResponseArray, kernelResponse)
+	serviceResponseArray = append(serviceResponseArray, osResponse)
```

Comment:
you can use 
`serviceResponseArray = append(serviceResponseArray, kernelResponse, osResponse)`
to append both response items